printing/printer.ml

```diff
@@ -828,9 +830,24 @@ let pr_goal_by_id ~proof id =
   try
     Proof.in_proof proof (fun sigma ->
       let g = Evd.evar_key id sigma in
-      pr_selected_subgoal (pr_id id) sigma g)
+      pr_selected_subgoal None (pr_id id) sigma g)
   with Not_found -> user_err Pp.(str "No such goal.")
 
+(** print a goal identified by the goal id as it appears in -emacs mode.
+ sid should be the Stm state id corresponding to proof.
+ Used to support the Prooftree tool in Proof General. (https://askra.de/software/prooftree/). *)
+let pr_goal_emacs ~proof gid sid =
+  match proof with
+  | None -> user_err Pp.(str "No proof for that state.")
+  | Some proof ->
+    let goals = Proof.all_goals proof in
+    try
+      let goal = Goal.Set.find (Evar.unsafe_of_int gid) goals in
+      Proof.in_proof proof (fun sigma ->
+          Pp.(pr_selected_subgoal (Some (str "goal ID " ++ (int gid) ++ str " at state " ++ (int sid)))
+            (str (string_of_int gid)) sigma goal))
+    with Not_found -> user_err Pp.(str "No such goal.")
+
```

Comment:
Well indeed it seems that you have added back the wrong command, as @hendriktews mentioned the old command was not doing this but printing evars [which seems like the right tool for their use case]

Code is here:
https://github.com/coq/coq/pull/765/files#diff-a77b692096a21947141aae664ef6812bL68

Reply:
In particular #6063 finished the job.

Reply:
This is what he asked for.  See https://github.com/coq/coq/issues/10315

Reply:
> This is what he asked for. See #10315

The command he asked for is not the one being bring back to life in this PR. And that seems as problem as the last comment in #10315 implies.

Reply:
Read the description at the top.  The discussion got off track on a request for an apparently unrelated feature.  We'll see what @hendriktews says.  This was supposed to be a very simple fix and it's becoming complex and confused.

Reply:
Maybe you should read the PRs removing the feature, the code doesn't match.